node/network/availability-distribution/src/tests/state.rs

```diff
@@ -0,0 +1,317 @@
+			match msg {
+				AllMessages::NetworkBridge(NetworkBridgeMessage::SendRequests(reqs)) => {
+					for req in reqs {
+						// Forward requests:
+						let in_req = to_incoming_req(&executor, req);
+
+						executor.spawn("Request forwarding",
+									overseer_send(
+										tx.clone(),
+										AvailabilityDistributionMessage::AvailabilityFetchingRequest(in_req)
+									).boxed()
+						);
+					}
+				}
+				AllMessages::AvailabilityStore(AvailabilityStoreMessage::QueryChunk(candidate_hash,	validator_index, tx)) => {
+					let chunk = self.chunks.get(&(candidate_hash, validator_index));
+					tx.send(chunk.map(Clone::clone))
+						.expect("Receiver is expected to be alive");
+				}
+				AllMessages::AvailabilityStore(AvailabilityStoreMessage::StoreChunk{candidate_hash,	chunk, tx, ..}) => {
+					assert!(
+						self.valid_chunks.contains(&(candidate_hash, chunk.index)),
+						"Only valid chunks should ever get stored."
+					);
+					tx.send(Ok(()))
+						.expect("Receiver is expected to be alive");
+					tracing::trace!(target: LOG_TARGET, "'Stored' fetched chunk.");
+					remaining_stores -= 1;
+				}
+				AllMessages::RuntimeApi(RuntimeApiMessage::Request(hash, req)) => {
+					match req {
+						RuntimeApiRequest::SessionIndexForChild(tx) => {
+							// Always session index 1 for now:
+							tx.send(Ok(1))
+							.expect("Receiver should still be alive");
+						}
+						RuntimeApiRequest::SessionInfo(_, tx) => {
+							tx.send(Ok(Some(self.session_info.clone())))
+							.expect("Receiver should be alive.");
+						}
+						RuntimeApiRequest::AvailabilityCores(tx) => {
+							tracing::trace!(target: LOG_TARGET, cores= ?self.cores[&hash], hash = ?hash, "Sending out cores for hash");
+							tx.send(Ok(self.cores[&hash].clone()))
+							.expect("Receiver should still be alive");
+						}
+						_ => {
+							panic!("Unexpected runtime request: {:?}", req);
+						}
+					}
+				}
+				_ => {
+					panic!("Unexpected message received: {:?}", msg);
+				}
+			}
+		}
+	}
+}
+
+
+
+async fn overseer_signal(
+	mut tx: SingleItemSink<FromOverseer<AvailabilityDistributionMessage>>,
+	msg: impl Into<OverseerSignal>,
+) {
+	let msg = msg.into();
+	tracing::trace!(target: LOG_TARGET, msg = ?msg, "sending message");
+	tx.send(FromOverseer::Signal(msg))
+		.await
+		.expect("Test subsystem no longer live");
+}
+
+async fn overseer_send(
+	mut tx: SingleItemSink<FromOverseer<AvailabilityDistributionMessage>>,
+	msg: impl Into<AvailabilityDistributionMessage>,
+) {
+	let msg = msg.into();
+	tracing::trace!(target: LOG_TARGET, msg = ?msg, "sending message");
+	tx.send(FromOverseer::Communication { msg }).await
+		.expect("Test subsystem no longer live");
+	tracing::trace!(target: LOG_TARGET, "sent message");
+}
+
+
+async fn overseer_recv(
+	rx: &mut mpsc::UnboundedReceiver<AllMessages>,
+) -> AllMessages {
+	tracing::trace!(target: LOG_TARGET, "waiting for message ...");
+	rx.next().await.expect("Test subsystem no longer live")
```

Comment:
we use timeouts from `subsystem-util` in other tests, which you may find useful here too

Reply:
Thanks @ordian! I use them already [here](https://github.com/paritytech/polkadot/blob/ccdd593b53fef16a11b33cd99ae3251960138598/node/network/availability-distribution/src/tests/state.rs#L154). Where else you think they might be beneficial?

Reply:
we use them at individual `recv`s, e.g. https://github.com/paritytech/polkadot/blob/652aad1b5c6d7ccebe4cd911c5c2b2825bafc1ba/node/core/av-store/src/tests.rs#L202

but I guess it doesn't really matter if the tests are passing, it just might be slightly more helpful to identify the issue if they don't

Reply:
Nice, thanks for the tip!